OADP-4058: Adding structure for OADP and 3scale user story (#85994)
openshift/openshift-docs: modules/restoring-the-secrets-and-apimanager.adoc

:_mod-docs-content-type: PROCEDURE

//included in backing-up-and-restoring-3scale-using-oadp.adoc assembly

[id="restoring-the-secrets-and-apimanager_{context}"]
= Restoring the secrets and APIManager

> 

Reviewer: needs summary sentence

.Prerequisites

> . You have backed up the 3scale Operator.

Reviewer: ```suggestion
. You backed up the 3scale Operator.
```

> . You have backed up mysql and Redis databases.

Reviewer: ```suggestion
. You backed up mysql and Redis databases.
```

.Procedure

> . Delete the 3scale operator CRDs along with the "threescale" namespace:

Reviewer: ```suggestion
. Delete the 3scale Operator custom resource definitions (CRDs) along with the "threescale" namespace by running the following command:
```

+
[source,terminal]
----
$ oc delete project threescale

> 

Reviewer: separate input v output

"threescale" project deleted successfully
----
+

> NOTE: If restoring on a separate cluster, ensure that {oadp-short} is installed and configured with `nodeAgent` enabled on the destination cluster similar to the source cluster.

Reviewer: properly format admonition and determine proper admonition type
```suggestion
[IMPORTANT]
====
If you are restoring the database on a different cluster than you backed up, ensure that {oadp-short} is installed and configured with `nodeAgent` enabled on the destination cluster as it was on to the source cluster.
====
```

> . Restore the Operator if you backed up the Operator. Otherwise, install a fresh 3scale operator:

Reviewer: Which is this operation? And where is the relevant command? This should be two steps, IMHO. Something like, 
. Get the 3scale Operator running by either restoring a backup or installing a new instance:

.. restore step
etc
.. install step
etc

+
[source,yaml]
----
apiVersion: velero.io/v1
kind: Restore
metadata:
  creationTimestamp: "2024-09-24T05:43:20Z"
  finalizers:
  - restores.velero.io/external-resources-finalizer
  generation: 6
  name: operator-installation-restore
  namespace: openshift-adp
  resourceVersion: "83988"
  uid: cbc10cd0-de8c-442d-bf29-40a07be70626
spec:
  backupName: operator-install-backup
  excludedResources:
  - nodes
  - events
  - events.events.k8s.io
  - backups.velero.io
  - restores.velero.io
  - resticrepositories.velero.io
  - csinodes.storage.k8s.io
  - volumeattachments.storage.k8s.io
  - backuprepositories.velero.io
  itemOperationTimeout: 4h0m0s
----

> . Manually create the `s3-credentials` secret: 

Reviewer: ```suggestion
. Manually create the `s3-credentials` Secret by using the following YAML example: 
```

+
[source,yaml]
----
apiVersion: v1
kind: Secret
metadata:

>        creationTimestamp: null

Reviewer: ```suggestion
       creationTimestamp: null
```
null?

       name: s3-credentials
stringData:

>   AWS_ACCESS_KEY_ID: something

Reviewer: ```suggestion
  AWS_ACCESS_KEY_ID: <access_key> # <1>
```

>   AWS_SECRET_ACCESS_KEY: something

Reviewer: ```suggestion
  AWS_SECRET_ACCESS_KEY: <aws_access_key> # <2>
```

>   AWS_BUCKET: something

Reviewer: ```suggestion
  AWS_BUCKET: <aws_bucket> # <3>
```

Author: Keeping this suggestion as it is, I've asked QE to find the real-life examples. I'll add its explanation using <1>, <2>, and such.

Author: It is now fixed

  AWS_REGION: us-east-1
type: Opaque
----

> 

Reviewer: And define these fields...will the user always use AWS to spin up a cluster, or should there be a more generic example here?
<1> 
<2>
<3>

> . Scale down the 3scale-operator:

Reviewer: ```suggestion
. Scale down the 3scale Operator by running the following command:
```

+
[source,terminal]
----

> # oc scale deployment threescale-operator-controller-manager-v2 --replicas=0 -n threescale

Reviewer: ```suggestion
$ oc scale deployment threescale-operator-controller-manager-v2 --replicas=0 -n threescale
```

Reviewer: Generally, we should not be suddenly from root. `oc` commands should be run by the cluster admin with their access, FWIW. Confirm with your SME and See https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#signs-and-symbols-in-code-blocks

----
+
[source,yaml]
----
Restore the Secrets:
apiVersion: velero.io/v1
kind: Restore
metadata:
  creationTimestamp: "2024-09-24T05:54:45Z"
  finalizers:
  - restores.velero.io/external-resources-finalizer
  generation: 6
  name: operator-resources-secrets
  namespace: openshift-adp
  resourceVersion: "89119"
  uid: 063e448d-2c71-4756-8828-eeab9c111b73
spec:
  backupName: operator-resources-secrets
  excludedResources:
  - nodes
  - events
  - events.events.k8s.io
  - backups.velero.io
  - restores.velero.io
  - resticrepositories.velero.io
  - csinodes.storage.k8s.io
  - volumeattachments.storage.k8s.io
  - backuprepositories.velero.io
  itemOperationTimeout: 4h0m0s
status:
  completionTimestamp: "2024-09-24T05:54:46Z"
  hookStatus: {}
  phase: Completed
  progress:
    itemsRestored: 13
    totalItems: 13
  startTimestamp: "2024-09-24T05:54:45Z"
----

> . Restore API Manager:

Reviewer: ```suggestion
. Restore the API manager by using the following YAML:
```
Are we missing an apply command?

+
[source,yaml]
----
apiVersion: velero.io/v1
kind: Restore
metadata:
  creationTimestamp: "2024-09-24T05:59:39Z"
  finalizers:
  - restores.velero.io/external-resources-finalizer
  generation: 6
  name: operator-resources-apim
  namespace: openshift-adp
  resourceVersion: "90854"
  uid: 2ebf0e2e-46bd-4ace-b78c-4d60e546bc3d
spec:
  backupName: operator-resources-apim
  excludedResources:
  - nodes
  - events
  - events.events.k8s.io
  - backups.velero.io
  - restores.velero.io
  - resticrepositories.velero.io
  - csinodes.storage.k8s.io
  - volumeattachments.storage.k8s.io
  - backuprepositories.velero.io
  itemOperationTimeout: 4h0m0s
status:
  completionTimestamp: "2024-09-24T05:59:40Z"
  hookStatus: {}
  phase: Completed
  progress:
    itemsRestored: 1
    totalItems: 1
  startTimestamp: "2024-09-24T05:59:39Z"
----

> . Scale up the 3scale-operator:

Reviewer: ```suggestion
. Scale up the 3scale Operator by running the following command:
```

+
[source,terminal]
----

> # oc scale deployment threescale-operator-controller-manager-v2 --replicas=1 -n threescale

Reviewer: ```suggestion
$ oc scale deployment threescale-operator-controller-manager-v2 --replicas=1 -n threescale
```

:_mod-docs-content-type: PROCEDURE

//included in backing-up-and-restoring-3scale-using-oadp.adoc assembly

[id="restoring-the-secrets-and-apimanager_{context}"]
= Restoring the secrets and APIManager

.Prerequisites

. You backed up the 3scale Operator.
. You backed up `mysql` and Redis databases.

.Procedure

> . Delete the 3scale Operator custom resource definitions (CRDs) along with the "threescale" namespace by running the following command:

Reviewer 2: ```suggestion
. Delete the 3scale Operator custom resource definitions (CRDs) along with the "threescale" namespace:
```

+
[source,terminal]
----
$ oc delete project threescale
----
+
.Example output:
+
[source,terminal]
----
"threescale" project deleted successfully
----
+
[IMPORTANT]
====

> If you are restoring the database on a different cluster than you backed up, ensure that {oadp-short} is installed and configured with `nodeAgent` enabled on the destination cluster as it was on to the source cluster.

Reviewer 2: A bit hard to understand..Is there a better way to say the same? 
I'd also advise to define `nodeAgent`.

Author: Added it to prerequisite - Ensure that you are restoring the database on the same cluster, where it was backed up. If it is on a different cluster, install and configure {oadp-short} with `nodeAgent` enabled on the destination cluster as it was on to the source cluster.

====

. Create the following YAML file to restore the 3scale Operator:
+
.Example restore.yaml file:
+
[source,yaml]
----
apiVersion: velero.io/v1
kind: Restore
metadata:
  name: operator-installation-restore
  namespace: openshift-adp
spec:
  backupName: operator-install-backup
  excludedResources:
  - nodes
  - events
  - events.events.k8s.io
  - backups.velero.io
  - restores.velero.io
  - resticrepositories.velero.io
  - csinodes.storage.k8s.io
  - volumeattachments.storage.k8s.io
  - backuprepositories.velero.io
  itemOperationTimeout: 4h0m0s
----

> . Restore the 3scale Operator by running the following command:

Reviewer 2: ```suggestion
. Restore the 3scale Operator:
```

+
[source,terminal]
----
$ oc create -f restore.yaml
----

> . Manually create the `s3-credentials` Secret by running the following command:

Reviewer 2: ```suggestion
. Manually create the `s3-credentials` Secret:
```

+
[source,terminal]
----
$ oc apply -f - <<EOF
---
apiVersion: v1
kind: Secret
metadata:
      name: s3-credentials
      namespace: threescale
stringData:
  AWS_ACCESS_KEY_ID: something
  AWS_SECRET_ACCESS_KEY: something
  AWS_BUCKET: something
  AWS_REGION: us-east-1
type: Opaque
EOF
----

> . Scale down the 3scale Operator by running the following command:

Reviewer 2: ```suggestion
. Scale down the 3scale Operator:
```

+
[source,terminal]
----
$ oc scale deployment threescale-operator-controller-manager-v2 --replicas=0 -n threescale
----

. Create the following example YAML file to restore Secrets:
+
.Example restore-secret.yaml file:
+
[source,yaml]
----
Restore the Secrets:
apiVersion: velero.io/v1
kind: Restore
metadata:
  name: operator-resources-secrets
  namespace: openshift-adp
spec:
  backupName: operator-resources-secrets
  excludedResources:
  - nodes
  - events
  - events.events.k8s.io
  - backups.velero.io
  - restores.velero.io
  - resticrepositories.velero.io
  - csinodes.storage.k8s.io
  - volumeattachments.storage.k8s.io
  - backuprepositories.velero.io
  itemOperationTimeout: 4h0m0s
----

> . Restore the Secrets by running the following command:

Reviewer 2: ```suggestion
. Restore the Secrets:
```

:_mod-docs-content-type: PROCEDURE

//included in backing-up-and-restoring-3scale-by-using-oadp.adoc assembly

[id="restoring-the-secrets-and-apimanager_{context}"]
= Restoring the secrets and APIManager

You can restore the secrets and APIManager by using the following procedure.

.Prerequisites

> . You backed up the 3scale Operator.

Author: Since it must be executed in the sequence, prerequisites are having an ordered list instead of bullets.
@bergerhoffer, is there any recommendation for Prerequisites for OCP for using ordered or unordered list?
Thanks in advance!

.Procedure

. Delete the 3scale Operator custom resource definitions (CRDs) along with the "threescale" namespace:
+
[source,terminal]
----
$ oc delete project threescale
----
+
.Example output:
+
[source,terminal]
----
"threescale" project deleted successfully
----

. Create a YAML file with the following configuration to restore the 3scale Operator:
+
.Example restore.yaml file:
+
[source,yaml]
----
apiVersion: velero.io/v1
kind: Restore
metadata:
  name: operator-installation-restore
  namespace: openshift-adp
spec:
  backupName: operator-install-backup
  excludedResources:
  - nodes
  - events
  - events.events.k8s.io
  - backups.velero.io
  - restores.velero.io
  - resticrepositories.velero.io
  - csinodes.storage.k8s.io
  - volumeattachments.storage.k8s.io
  - backuprepositories.velero.io
  itemOperationTimeout: 4h0m0s
----

. Restore the 3scale Operator:
+
[source,terminal]
----
$ oc create -f restore.yaml
----

. Manually create the `s3-credentials` Secret:
+
[source,terminal]
----

> $ oc apply -f - <<EOF

Author: It was suggested by the SME for this file but I can check it again.
